disk-utils/fsck.c

```diff
@@ -819,10 +819,10 @@ static struct fsck_instance *wait_one(int flags)
 		for (inst2 = instance_list; inst2; inst2 = inst2->next) {
 			if (inst2->flags & FLAG_DONE)
 				continue;
-			if (strcmp(inst2->type, "ext2") &&
+			if (strcmp(inst2->type, "ext2") != 0 &&
 			    strcmp(inst2->type, "ext3") &&
```

Comment:
Is it really really necessary to "fix" this issue? I guess we're smart enough to understand the code.

Reply:
well, != 0 is not necessary, but it is recommended. Hence why it's listed under bugprone.

Reply:
OK :-)